cli/azd/pkg/account/manager.go

```diff
@@ -0,0 +1,250 @@
+package account
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+	"golang.org/x/exp/slices"
+)
+
+// Manages azd account configuration
+// `az cli` is not required and will only be called `azd` default have not already been set.
+type Manager struct {
+	azCli azcli.AzCli
+}
+
+// Creates a new Account Manager instance
+func NewManager(ctx context.Context) *Manager {
+	return &Manager{
+		azCli: azcli.GetAzCli(ctx),
+	}
+}
+
+// Gets the default subscription for the logged in account.
+// 1. Returns AZD config defaults if exists
+// 2. Returns AZ CLI defaults if exists
+func (m *Manager) GetAccountDefaults(ctx context.Context) (*config.Config, error) {
+	azdConfig := config.GetConfig(ctx)
+
+	if azdConfig.DefaultSubscription != nil && azdConfig.DefaultLocation != nil {
+		return azdConfig, nil
+	}
+
+	if azdConfig.DefaultSubscription == nil {
+		defaultSubscription, err := m.getDefaultSubscription(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed retrieving default subscription from Azure CLI: %w", err)
+		}
+
+		azdConfig.DefaultSubscription = defaultSubscription
+	}
+
+	if azdConfig.DefaultLocation == nil {
+		defaultLocation, err := m.getDefaultLocation(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed retrieving default location from Azure CLI: %w", err)
+		}
+
+		azdConfig.DefaultLocation = defaultLocation
+	}
+
+	return azdConfig, nil
+}
+
+// Gets the available Azure subscriptions for the current logged in account.
+func (m *Manager) GetSubscriptions(ctx context.Context) ([]azcli.AzCliSubscriptionInfo, error) {
```

Comment:
Seems like this class already handles "get current account information", maybe we should move subscription and location listing somewhere else